tests/aggregation/tests.py

```diff
@@ -2158,6 +2159,85 @@ def test_having_with_no_group_by(self):
         )
         self.assertEqual(list(author_qs), [337])
 
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg(self):
+        vals = Store.objects.aggregate(jsonarrayagg=JSONArrayAgg("name"))
+        self.assertEqual(
+            vals,
+            {"jsonarrayagg": ["Amazon.com", "Books.com", "Mamma and Pappa's Books"]},
+        )
+
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg_datefield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("pubdate"))
+        self.assertEqual(
+            vals,
+            {
+                "jsonarrayagg": [
+                    "2007-12-06",
+                    "2008-03-03",
+                    "2008-06-23",
+                    "2008-11-03",
+                    "1995-01-15",
+                    "1991-10-15",
+                ]
+            },
+        )
+
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg_decimalfield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("price"))
+        self.assertEqual(
+            vals, {"jsonarrayagg": [30.0, 23.09, 29.69, 29.69, 82.8, 75.0]}
+        )
+
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg_integerfield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("pages"))
+        self.assertEqual(vals, {"jsonarrayagg": [447, 528, 300, 350, 1132, 946]})
+
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg_filter(self):
+        vals = Author.objects.aggregate(
+            jsonarrayagg=JSONArrayAgg("age", filter=Q(age__gt=29))
+        )
+
+        self.assertEqual(vals, {"jsonarrayagg": [34, 35, 45, 37, 57, 46]})
+
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg_empty_result_set(self):
+        Author.objects.all().delete()
+
+        val = Author.objects.aggregate(jsonarrayagg=JSONArrayAgg("age"))
+
+        self.assertEqual(val, {"jsonarrayagg": None})
+
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg_default_set(self):
+        Author.objects.all().delete()
+
+        val = Author.objects.aggregate(
+            jsonarrayagg=JSONArrayAgg("name", default=["<empty>"])
+        )
+        self.assertEqual(val, {"jsonarrayagg": ["<empty>"]})
+
+    @skipUnlessDBFeature("supports_aggregate_filter_clause")
+    def test_JSONArrayAgg_distinct_false(self):
```

Comment:
Actually I'm not sure this test adds any value since `distinct=False` is the default so this behavior is implicitly tested in other instances.